twilight-gateway/src/ratelimiter.rs

```diff
@@ -185,7 +190,7 @@ mod tests {
 
         assert_eq!(ratelimiter.available(), ratelimiter.max());
         for _ in 0..ratelimiter.max() {
-            ratelimiter.acquire().await;
+            poll_fn(|cx| ratelimiter.poll_acquire(cx)).await;
```

Comment:
consider making `acquire` a helper function? I'm not a big fan of repetitive `poll_fn` usage, it's a big ugly in my view

Reply:
I'd rather not add test-only helper methods